### PR TITLE
Don't crash when non-object formData is passed in to a schema item with additionalProperties

### PR DESCRIPTION
**Pull request by olensmar**
### Reasons for making this change

Fixed a crash when null formData was passed for property with additionalProperties defined in the schema

### Checklist

* [ ] **I'm updating documentation**
  - [ ] I've [checked the rendering](https://react-jsonschema-form.readthedocs.io/en/latest/#contributing) of the Markdown text I've added
* [X] **I'm adding or updating code**
  - [X] I've added and/or updated tests
  - [ ] I've updated [docs](https://react-jsonschema-form.readthedocs.io/) if needed
  - [ ] I've updated the [changelog](https://github.com/rjsf-team/react-jsonschema-form/blob/master/CHANGELOG.md) with a description of the PR
* [ ] **I'm adding a new feature**
  - [ ] I've updated the playground with an example use of the feature

### CODE REVIEW

File: packages/core/src/utils.js [hunk 1]
```diff
@@ -634,6 +634,9 @@ export function stubExistingAdditionalProperties(
     properties: { ...schema.properties },
   };
 
+  // handle null formData
+  formData = formData || {};
```
Comment (epicfaace) [inline packages/core/src/utils.js:638]: Shouldn't `formData = {}` already fix this on line 629? Why doesn't that work?

Comment (epicfaace) [reply packages/core/src/utils.js:638]: I'm trying to understand when `formData` would ever be `null`? Won't it always be an object or `undefined`?

Comment (olensmar) [reply packages/core/src/utils.js:638]: Hi @epicfaace - thanks for looking into this! 

I bumped into this where I had a nested object property that is explicitly set to null (not undefined) which caused the error you will get if you comment my fix and run the provided test..

The actual code in the npm package for `stubExistingAdditionalProperties` contains

```javascript
var formData = arguments.length > 2 && arguments[2] !== undefined ? arguments[2] : {};
```

which you can see doesn't handle null - only undefined. I think better would be 

```javascript
var formData = arguments.length > 2 && arguments[2] ? arguments[2] : {};
```

since null evaluates to false - but I don't know where this "transpilation" is coming from!? 

Happy to provide more context/info if needed!

Comment (epicfaace) [reply packages/core/src/utils.js:638]: The transpilation happens because the original code says

```
export function stubExistingAdditionalProperties(
  schema,
  rootSchema = {},
  formData = {}
) 
```

Specifying a default argument of `{}` means that the default will be reverted to only if `formData` is undefined (but not if it is null).



> I bumped into this where I had a nested object property that is explicitly set to `null` (not `undefined`) which caused the error you will get if you comment my fix and run the provided test..

Interesting! Can you add a test that explicitly checks for this condition from the end user's point of view? i.e. a test in Form_test.js that sets the formData according to what you mentioned

Comment (olensmar) [reply packages/core/src/utils.js:638]: Hi @epicfaace - sorry for the delay - i've added the corresponding Form test - if you uncomment my fix you should get something like the following error when running `npm run test` 
```
@rjsf/core:   5) utils
@rjsf/core:        retrieveSchema()
@rjsf/core:          should handle null formData for schema which contains additionalProperties:
@rjsf/core:      TypeError: Cannot convert undefined or null to object
@rjsf/core:       at Function.keys (<anonymous>)
@rjsf/core:       at stubExistingAdditionalProperties (src/utils.js:674:10)
@rjsf/core:       at retrieveSchema (src/utils.js:758:12)
```

Hope this helps!

Comment (epicfaace) [reply packages/core/src/utils.js:638]: Thanks for the test, I get what you're doing now. I think we want to handle not only null formData, but any non-object formData.

How about we do this instead:

```suggestion
formData = isObject(formData) ? formData : {};
```

And then let's also add an additional test where the formData is set to a non-null non-object (for example, when formData is equal to `2` or an array)?

File: packages/core/test/Form_test.js [hunk 2]
```diff
@@ -506,6 +506,29 @@ describeRepeated("Form common", createFormComponent => {
       expect(node.querySelectorAll("input[type=text]")).to.have.length.of(1);
     });
 
+    it("should handle null values for property with additionalProperties", () => {
```
Comment (epicfaace) [inline packages/core/test/Form_test.js:509]:
```suggestion
    it("should not crash with null values for property with additionalProperties", () => {
```

Comment (olensmar) [reply packages/core/test/Form_test.js]: done!

File: packages/core/test/utils_test.js [hunk 3]
```diff
@@ -1770,6 +1770,21 @@ describe("utils", () => {
       });
     });
 
+    it("should handle null formData for schema which contains additionalProperties", () => {
```
Comment (olensmar) [inline packages/core/test/utils_test.js:1773]: @epicfaace do you still want to keep this test?

**Comment by olensmar** [comment]
thanks @epicfaace - i've made some more updates in line with your suggestions

File: packages/core/test/utils_test.js (hunk 3 above)
Comment (epicfaace) [reply packages/core/test/utils_test.js:1773]: Sure

**Review by epicfaace — APPROVED** [review]
Thanks! This is great.

**Comment by olensmar** [comment]
my pleasure - thanks for a great project!

**Comment by epicfaace** [comment]
Can you fix formatting (see CI)?

**Comment by olensmar** [comment]
@epicfaace done! (I think...)

**Comment by olensmar** [comment]
thanks again @epicfaace - when do you expect to do a release including this fix?

**Comment by epicfaace** [comment]
doing one now
